src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -347,7 +352,7 @@ def _reset_eval_dataloader(
             dataloaders = self._resolve_overfit_batches(dataloaders, mode)
 
         if not isinstance(dataloaders, list):
-            dataloaders = [dataloaders]
+            dataloaders = [dataloaders]  # type: ignore[list-item]
```

Comment:
associated error ignored:

```sh
src/pytorch_lightning/trainer/connectors/data_connector.py:355: error: List item 0 has incompatible type "Union[DataLoader[Any], Collection[DataLoader[Any]]]"; expected "DataLoader[Any]"  [list-item]
```

Reply:
This is not a proper way, how to resolve this. I'd rather see it resolved in methods generating this value (i.e. `self._resolve_overfit_batches` and `self._request_dataloader`)

Reply:
`_resolve_overfit_batches` is [used in trainer](https://github.com/Lightning-AI/lightning/blob/4e9fbdbe4013458ea33256f4e0b00cf0782e6064/src/pytorch_lightning/trainer/trainer.py#L1813) in a way that does not require a list be returned; updating `_resolve_overfit_batches` to fix the error here, would require that I alter code in trainer.

I tried `dataloaders = list(dataloaders)` out of curiosity, and that worked locally.

Updated in https://github.com/Lightning-AI/lightning/pull/13603/commits/ca4f1609731b64201c3fe184fe71a104fd15276d

Reply:
But `list(dataloaders) != [dataloaders]`. `list(iterable)` tries to convert that iterable into list (which would succeed for single dataloader, but is definitely not something we want here), whereas `[iterable]` creates a list with `iterable` as a first element. No wonder mypy stopped complaining, other tests had to start complaining.

Reply:
Anyway, after taking a look that changing the methods is probably not feasible within this PR, I think in this case it makes sense to again, `# type: ignore`

Reply:
Sorry about this

Reply:
reverted in https://github.com/Lightning-AI/lightning/pull/13806/commits/c4039b1bc929aeb96049cfbd85e3b032f2166f14